### PR TITLE
Build eglib under none desktop Windows API family.

### PR DESCRIPTION
**Pull request by lateralusX**
Initial work to build eglib under none desktop Windows API families. There will be more PR's comming building on this initial PR for other mono runtime libraries.

The classic Windows API have been split into families and partitions in order to support
different subsets off the Windows API on different platforms. In order to build against other Windows targets/platforms, Mono needs to follow these families/partitions in order to successfully build where API’s have been removed/replaced/extended. Since most Mono platforms uses the Windows API signatures, a new defined has been added to all platforms, HAVE_CLASSIC_WINAPI_SUPPORT, used to decide what implementation to include. On none Windows platforms this is always defined and on Windows platforms it follows the winapifamily.h setup in winconfig.h. In order to keep down regressions, the existing API’s are kept for HAVE_CLASSIC_WINAPI_SUPPORT even if there are more modern alternatives and replacements available.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -345,6 +345,9 @@ if test x$target_win32 = xyes; then
    AC_DEFINE(TARGET_WIN32, 1, [Target Platform is Win32])
 fi
 
+# Defined for all targets/platforms using classic Windows API support.
+AC_DEFINE(HAVE_CLASSIC_WINAPI_SUPPORT, 1, [Use classic Windows API support])
```
Comment (ntherning) [inline configure.ac]: I believe we should define the other `HAVE_<family>_WINAP_SUPPORT` defines to 0 here as well.

File: eglib/configure.ac [hunk 2]
```diff
@@ -130,6 +130,9 @@ AM_CONDITIONAL(TARGET_WIN32, test x$OS = xWIN32)
 AM_CONDITIONAL(PLATFORM_DARWIN, test x$platform_darwin = xyes)
 AM_CONDITIONAL(PLATFORM_ANDROID, test x$platform_android = xyes)
 
+# Defined for all targets/platforms using classic Windows API support.
+AC_DEFINE(HAVE_CLASSIC_WINAPI_SUPPORT, 1, [Use classic Windows API support])
```
Comment (ntherning) [inline eglib/configure.ac]: I believe we should define the other `HAVE_<family>_WINAP_SUPPORT` defines to 0 here as well.

**Comment by lateralusX** [comment]
Added explicit define setting HAVE_UWP_WINAPI_SUPPORT to 0 on all platforms. This imirrors how the defines are setup on windows platforms using classic winapi support:

#define HAVE_CLASSIC_WINAPI_SUPPORT 1
#define HAVE_UWP_WINAPI_SUPPORT 0

File: configure.ac (hunk 1 above)
Comment (lateralusX) [inline configure.ac]: OK

File: eglib/configure.ac (hunk 2 above)
Comment (lateralusX) [inline eglib/configure.ac]: OK

File: eglib/configure.ac [hunk 3]
```diff
@@ -130,6 +130,10 @@ AM_CONDITIONAL(TARGET_WIN32, test x$OS = xWIN32)
...
+AC_DEFINE(HAVE_UWP_WINAPI_SUPPORT, 0, [Don't use UWP Windows API support])
```
Comment (kumpera) [inline eglib/configure.ac]: At some point we should make this a configure time toggle so we can enable it on desktop and test it.

Comment (kumpera) [inline eglib/configure.ac]: oh, nm, just show winconfig

**Comment by kumpera** [comment]
LGTM

File: eglib/configure.ac (hunk 3 above)
Comment (lateralusX) [inline eglib/configure.ac]: Right now you can do this manually (by changing winconfig.h) and this is how it is currently tested. Windows builds have never used the result from configure steps, it uses the checked in version of winconfig.h. This file uses the winfamilyapi.h to set the right API family based on build targets and it is possible to manually override just to test mono API Surface.

**Pull request merged**